### PR TITLE
feat(catalog): ResolveBangumi exact-first title resolution (CATALOG-1 #941)

### LINKED ISSUE #941: CATALOG-1 — Resolve Bangumi titles exact-first
## Parent

#936

## Spec card

`CATALOG-1`

## What to build

ResolveBangumi executes exact-first title resolution through Neon aliases and the explicit upstream-ingest adapter.

Migrate its published route and Agent gateway capability; delete mixed transport/SQL/upstream orchestration, shadow DTO, and pass-through tests.

## Observability

Record typed outcome, candidate count, source class, and duration without query text or upstream body.

## Acceptance criteria

- [ ] Exact, alias, ambiguity, not-found, ingest, and upstream failure pass; weakening exact-first resolution fails. [integration]

## Delivery contract

One dedicated worktree and one reviewable PR. Begin with the behavior manifest and characterization commit; land the real local/staging adapter, caller migration, old-path deletion, redacted observability, named mutation proof, and Codecov patch coverage of at least 95%. No compatibility layer.

## Blocked by

- #937 (SAFE-1)

### PR DESCRIPTION
**Pull request by lifeodyssey**
Closes #941 (parent #936). Wave 1.

- `ResolveBangumi` use case: exact-first sequencing via `TitleAliasPort` (Neon aliases) + `UpstreamTitlePort` (ingest), typed outcomes (resolved/needs_disambiguation/not_found/upstream_unavailable), redacted observation (outcome/candidate_count/source_class/duration).
- Route + agent gateway compose the use case; deleted mixed orchestration (`src/api/resolve.ts`), shadow DTOs (`AliasWork`/`ResolveDb`), pass-through tests.
- Red tests written first (7 behaviors incl. exact-first ordering); 2 named mutations killed (upstream-before-exact, swallowed upstream failure).
- Gates: typecheck, oxlint deny-warnings, 45 files/334 tests, coverage 95.16/82.77/98.44/97.17, spike 31 pass, startup smoke 200, contract parity green.

## Summary by Sourcery

Introduce a deterministic ResolveBangumi use case that resolves anime titles via an exact-first alias index, falling back to Bangumi upstream search with typed outcomes and redacted observability.

New Features:
- Expose ResolveBangumi as an application-level resolver that sequences alias lookup before upstream title search.
- Add TitleAliasPort and UpstreamTitlePort outbound interfaces, with Neon-backed alias lookup and Bangumi search adapters.
- Record resolve observations (outcome, candidate count, source class, duration) through a pluggable observer without logging query or upstream payloads.

Enhancements:
- Refactor the catalog resolve route to compose the new ResolveBangumi use case and ports instead of the previous mixed SQL/fetch orchestration.
- Export a reusable candidateFromRow helper to consistently derive anime candidates from Bangumi rows.

Tests:
- Replace previous resolve DB and miss/hit tests with focused suites covering exact-first alias behavior, upstream similarity and failure handling, adapter wiring, and route-level validation and outcomes.

## Summary by CodeRabbit

* **New Features**
  * Added title resolution using exact aliases, upstream search, candidate ranking, deduplication, and result limits.
  * Added fallback handling for unavailable upstream search services.
  * Added enriched candidate information from stored catalog data.
  * Added redacted resolution outcome and timing telemetry.

* **Bug Fixes**
  * Improved handling of ambiguous aliases, orphaned entries, malformed results, and empty searches.

* **Tests**
  * Expanded coverage for successful resolutions, fallbacks, validation, ranking, upstream failures, and edge cases.

### CODE REVIEW

File: workers/catalog/test/resolve-bangumi.worker.test.ts [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+    expect(observed).toEqual([{
+      outcome: "resolved",
+      candidate_count: 1,
+      source_class: "alias",
+      duration_ms: 1,
+    }]);
+  });
+
+  it("records an upstream miss observation with source class upstream", async () => {
```
[the remark this answers is not in the document]
Comment (lifeodyssey) [reply workers/catalog/test/resolve-bangumi.worker.test.ts:168]: fixed in 22098b94.

File: workers/catalog/test/resolve-bangumi.worker.test.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+    const result = await resolveBangumi(aliasWith(works, candidates), upstreamWith([]), { query: "Series" });
+    expect(result.outcome).toBe("needs_disambiguation");
+    const ambiguity = result as Extract<typeof result, { outcome: "needs_disambiguation" }>;
+    expect(ambiguity.candidates).toHaveLength(MAX_CANDIDATES);
+    expect(ambiguity.candidates.map((item) => item.bangumi_id)).toEqual(["1", "2", "3", "4", "5", "6"]);
+  });
+});
+
+describe("resolveBangumi redacted observability", () => {
```
[the remark this answers is not in the document]
Comment (lifeodyssey) [reply workers/catalog/test/resolve-bangumi.worker.test.ts:148]: fixed in 22098b94.

File: workers/catalog/test/resolve-upstream.worker.test.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  return { id: String(id), name, name_cn };
+}
+
+const EMPTY_ALIAS: TitleAliasPort = {
+  worksForAlias: () => Promise.resolve([]),
+  candidatesForWorks: () => Promise.reject(new Error("catalog candidates must not load on MISS")),
+};
+
+describe("resolveBangumi upstream miss outcomes", () => {
+  it("falls through to upstream when every alias work is orphaned", async () => {
```
[the remark this answers is not in the document]
Comment (lifeodyssey) [reply workers/catalog/test/resolve-upstream.worker.test.ts:24]: fixed in 22098b94.

File: workers/catalog/src/adapters/outbound/title-alias.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  return result.rows.map(readAliasWork);
+}
+
+async function selectCandidates(db: AliasDb, workIds: string[]): Promise<AnimeCandidate[]> {
+  const result = await db.execute(sql`
+    SELECT b.id, b.title, b.title_cn, b.cover_url, b.air_date,
+           COUNT(p.id) AS points_count
+    FROM bangumi b LEFT JOIN points p ON p.bangumi_id = b.id
+    WHERE b.id IN (${sql.join(workIds, sql`, `)})
+    GROUP BY b.id, b.title, b.title_cn, b.cover_url, b.air_date
```
[the remark this answers is not in the document]
Comment (lifeodyssey) [reply workers/catalog/src/adapters/outbound/title-alias.ts:42]: fixed: parsedRow split under 10 lines (emptyEnrichment) — 22098b94.

File: workers/catalog/src/application/resolve-bangumi.ts [hunk 5]
```diff
@@ -22,41 +27,94 @@ const MIN_QUERY_LEN = 2;
...
 ): Promise<ResolveOutcome> {
-  const hit = await aliasHit(db, normalizeAlias(input.query));
-  if (hit) return hit;
-  return resolveMiss(input.query, opts.fetchImpl);
+  const clock = opts.clock ?? realClock;
+  const started = clock.now();
+  const result = await resolveExactFirst(alias, upstream, input.query);
+  opts.observer?.record(observe(result, started, clock.now()));
+  return result.outcome;
+}
```
[the remark this answers is not in the document]
Comment (lifeodyssey) [reply workers/catalog/src/application/resolve-bangumi.ts:95]: fixed in 22098b94.

File: workers/catalog/test/resolve-bangumi-guards.worker.test.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    const works: AliasWork[] = [
+      { bangumi_id: "100", priority: 40 },
+      { bangumi_id: "200", priority: 40 },
+    ];
+    const candidates = [candidate("100", 3), candidate("100", 3), candidate("200", 7)];
+
+    const result = await resolveBangumi(aliasWith(works, candidates), upstreamWith([]), { query: "Fate" });
+    expect(result.outcome).toBe("needs_disambiguation");
+    const ambiguity = result as Extract<typeof result, { outcome: "needs_disambiguation" }>;
+    expect(ambiguity.candidates.map((item) => item.bangumi_id)).toEqual(["200", "100", "100"]);
```
[the remark this answers is not in the document]
Comment (lifeodyssey) [reply workers/catalog/test/resolve-bangumi-guards.worker.test.ts:63]: fixed: duplicate-row ordering now asserted (identical-field candidates distinguished by id order) — 22098b94.
